src/components/tabController/TabBar.tsx

```diff
@@ -144,47 +139,27 @@ const TabBar = (props: Props) => {
     containerWidth: propsContainerWidth,
     centerSelected,
     containerStyle,
-    testID,
-    children: propsChildren
+    testID
   } = props;
 
   const context = useContext(TabBarContext);
   // @ts-ignore // TODO: typescript
   const {itemStates, items: contextItems, currentPage, targetPage, registerTabItems, selectedIndex} = context;
 
-  const children = useRef<Props['children']>(_.filter(propsChildren, (child: ChildProps) => !!child));
-
   const _registerTabItems = () => {
     const ignoredItems: number[] = [];
-    let itemsCount;
-
-    if (propsItems) {
-      itemsCount = _.size(propsItems);
-      _.forEach(propsItems, (item, index) => {
-        if (item.ignore) {
-          ignoredItems.push(index);
-        }
-      });
-      // TODO: deprecate with props.children
-    } else {
-      itemsCount = React.Children.count(children.current);
-      // @ts-ignore TODO: typescript - not sure if this can be solved
-      React.Children.toArray(children.current).forEach((child: ChildProps, index: number) => {
-        if (child.props.ignore) {
-          ignoredItems.push(index);
-        }
-      });
-    }
+    const itemsCount = _.size(propsItems);
+    _.forEach(propsItems, (item, index) => {
+      if (item.ignore) {
+        ignoredItems.push(index);
+      }
+    });
 
     registerTabItems(itemsCount, ignoredItems);
   };
 
   useEffect(() => {
-    if (propsChildren) {
-      LogService.warn('uilib: Please pass the "items" prop to TabController.TabBar instead of children');
-    }
-
-    if ((propsItems || children.current) && !contextItems) {
+    if (propsItems && !contextItems) {
       _registerTabItems();
     }
   }, []);
```

Comment:
Why no dependencies here, like 'items'? Won't it cause unnecessary updates?

Reply:
On the contrary, no dependencies means no re-renders (this is like `componentDidMount`).